lib/spack/spack/environment/environment.py

```diff
@@ -367,6 +383,14 @@ def create_in_dir(
 
     return env
 
+    # Must be done after environment is initialized
+    if include_concrete:
+        manifest.set_include_concrete(include_concrete)
+
+    if not keep_relative and init_file is not None and str(init_file).endswith(manifest_name):
+        init_file = pathlib.Path(init_file)
+        manifest.absolutify_dev_paths(init_file.parent)
```

Comment:
This code is unreachable, see:
```console
return env
```
above

Reply:
That looks like unresolved rebase cleanup.  I see if I can track down where the unreachable code is from and fix it.

Reply:
@tgamblin Did you see :point_up: This is very likely a bug?

Reply:
I checked a couple of commits back a couple of rebases and have not found where that was introduced yet.  Will proceed with the fix per Todd.

Reply:
Looks like it is likely rebase residual related to conflict resolution with #35273 .

Reply:
Removed in https://github.com/spack/spack/pull/44065.